provider/machinepool/machine_pool_resource.go

```diff
@@ -522,7 +522,7 @@ func (r *MachinePoolResource) readState(ctx context.Context, state *MachinePoolS
 	return
 }
 
-func assertNoChanges(state, plan *MachinePoolState) diag.Diagnostics {
+func validateNoImmutableAttChange(state, plan *MachinePoolState) diag.Diagnostics {
```

Comment:
I think it's bettek to have a name conversion - so if you change here the name, please update in all other resources...

Reply:
Done - change the function in cluster rosa resource as well